browser/ui/views/text_recognition_dialog_view.cc

```diff
@@ -140,5 +248,47 @@ void TextRecognitionDialogView::AdjustWidgetSize() {
   GetWidget()->SetSize(GetDialogClientView()->GetPreferredSize());
 }
 
+void TextRecognitionDialogView::OnShowResultTimerFired() {
+  // Fired before getting text from image.
+  // Will be updated when text is fetched.
+  if (!result_) {
+    return;
+  }
+
+  // Fired after getting text from image.
+  // Show the result now.
+  OnGetTextFromImage(*result_);
+}
+
+#if BUILDFLAG(IS_WIN)
+void TextRecognitionDialogView::OnGetAvailableRecognizerLanguages(
+    const std::vector<std::string>& languages) {
+  CHECK(content::BrowserThread::CurrentlyOn(content::BrowserThread::UI));
```

Comment:
nit: these CHECKs should be replaced with `SequenceChecker`. I think we shouldn't care about UI/not UI thread, but instead ensure the sequence used in `TextRecognitionDialogView` constructor is the same for "reply" calls.